### PR TITLE
Use PAL Robotics ddynamic_reconfigure 0.2.0

### PR DESCRIPTION
**Pull request by v-lopez**
ddynamic_reconfigure is being released here: https://github.com/ros/rosdistro/pull/20651

I've only been able to test it with our packages and with an Intel RealSense T265, and I was able to change the frames_queue_size param.

### CODE REVIEW

File: realsense2_camera/src/base_realsense_node.cpp [hunk 1]
```diff
@@ -278,7 +278,10 @@ void BaseRealSenseNode::registerDynamicOption(ros::NodeHandle& nh, rs2::options
             {
                 sensor.set_option(option, option_value);
             }
-            ddynrec->add(new DDBool(option_name, i, sensor.get_option_description(option), option_value));
+            ddynrec->registerVariable<bool>(
+              option_name, option_value,
+              [option, sensor](bool new_value) { sensor.set_option(option, new_value); },
```
Comment (efernandez) [inline realsense2_camera/src/base_realsense_node.cpp:283]: Probably not important, but maybe capture by reference here and in the other lambdas. Here it'd be: `[&option, &sensor]`

**Review by efernandez — APPROVED** [review]
LGTM, only a minor comment, but still needs to give it a try with my D435 sensor.

**Comment by efernandez** [comment]
The tests are failing because they can't find `ddynamic_reconfigure`. This is expected. Not sure if they'd pass once `ddynamic_reconfigure` release hits the rosdistro.

**Comment by doronhi** [comment]
ros-kinetic-ddynamic-reconfigure is now available.
I had an impression that we were talking about version 0.2.0. However, while installing it writes:
```
Setting up ros-kinetic-ddynamic-reconfigure (0.1.7-0xenial-20190320-153332-0800)
```
Is it a mistake somewhere...?

After this installation, building this PR fails for me.

Posting the first error line:
```
/home/doronhi/catkin_ws/src/realsense/realsense2_camera/src/base_realsense_node.cpp: In member function ‘void realsense2_camera::BaseRealSenseNode::registerDynamicOption(ros::NodeHandle&, rs2::options, std::__cxx11::string&)’:
/home/doronhi/catkin_ws/src/realsense/realsense2_camera/src/base_realsense_node.cpp:281:22: error: ‘class ddynamic_reconfigure::DDynamicReconfigure’ has no member named ‘registerVariable’
             ddynrec->registerVariable<bool>(
                      ^
/home/doronhi/catkin_ws/src/realsense/realsense2_camera/src/base_realsense_node.cpp:281:39: error: expected primary-expression before ‘bool’
             ddynrec->registerVariable<bool>(
                                       ^
/home/doronhi/catkin_ws/src/realsense/realsense2_camera/src/base_realsense_node.cpp:283:89: error: expected primary-expression before ‘,’ token
               [option, sensor](bool new_value) { sensor.set_option(option, new_value); },
                                                                                         ^
```
Could it be something that remains from the old version and I should try again on a new machine?

Two other points:
1. For the CI test to pass, the .travis.yaml file should be updated to install ros-kinetic-ddynamic-reconfigure.
2. I noticed the the "description" parameter is gone from the registerVariable function. Personally I liked that functionality. Do you plan on bringing it back?

**Comment by v-lopez** [comment]
No, you need version 0.2.0 at least for this.
The rosdistro MR was merged yesterday: https://github.com/ros/rosdistro/pull/20651

I don't know if it's still being processed and will be available soon.

The description parameter is there on 0.2.0

**Comment by v-lopez** [comment]
I requested 0.2.0 the day after  0.1.7, but it looks like only 0.1.7 got in the current sync, and 0.2.0 is waiting for the next one.

Maybe temporarily you want to include ddynamic_reconfigure 0.2.0 in this repo in the same way you had the old one.

**Comment by doronhi** [comment]
Thanks. I'll see my timelines and use what is available at the time.
what about the "description" field?

**Comment by v-lopez** [comment]
The description field is already implemented in [0.2.0](https://github.com/pal-robotics/ddynamic_reconfigure/blob/kinetic-devel/include/ddynamic_reconfigure/ddynamic_reconfigure.h#L45) and is being used in this MR [here](https://github.com/intel-ros/realsense/pull/693/commits/d9f8cb960e4dde597b1629e5b8f6bb2fdaef26c2#diff-15761e8f7813869ae417a96398deda78R312)

**Comment by doronhi** [comment]
I must have missed it somehow. Sorry and Thanks.

File: realsense2_camera/src/base_realsense_node.cpp (hunk 1 above)
Comment (v-lopez) [reply realsense2_camera/src/base_realsense_node.cpp:283]: I believe that when `option` and `sensor` go out of scope, the references will no longer be valid, and accessing them will cause undefined behavior.

**Comment by v-lopez** [comment]
`ddynamic_reconfigure` has been released at version 0.2.0 in kinetic. I've rebased this branch and added the missing install rule to the travis cfg.

File: realsense2_camera/src/base_realsense_node.cpp (hunk 1 above)
Comment (efernandez) [reply realsense2_camera/src/base_realsense_node.cpp:283]: True, these lambdas are used outside of this method scope.
I was also concerned about copying `sensor`, but it looks like this method already copies it, so it should be fine. :+1:

**Comment by doronhi** [comment]
Thanks!

**Comment by Karsten1987** [comment]
any plans of releasing this for melodic?

**Comment by v-lopez** [comment]
There's a PR for ddynamic_reconfigure here https://github.com/ros/rosdistro/pull/21380

**Pull request merged**